client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.js

```diff
@@ -11,122 +11,274 @@ import {
+      const data_tppc = std_years.map((year, i) => {
+        const single_year_tp_data = get_subject_data_for_year(year);
+        const result = _.chain(
+          _.keys(single_year_tp_data))
+          .pullAll(["na", "abroad"])
+          
+          .map((prov) => {
+            const in_year_prov_transfer_payments = single_year_tp_data[prov];
+            const in_year_prov_population = population[prov][i];
+            return [
+              prov,
+              in_year_prov_transfer_payments/in_year_prov_population,
+            ];
+          })
+          .fromPairs()
+          .value();
+        return result;
+      });
+      
+      //REGULAR TP VERSION
+      const current_year_data_tp = _.last(data_tp);
+      
+      const max_tp = _.chain(data_tp)
+        .last()
+        .values()
+        .max()
+        .value();
+      const color_scale_tp = d3.scaleLinear()
+        .domain([0, max_tp])
+        .range([0.2, 1]);
+
+      const largest_prov_tp = _.chain(current_year_data_tp)
         .keys()
-        .maxBy( (prov) => current_year_data[prov] )
+        .maxBy( (prov) => current_year_data_tp[prov] )
         .value();
-      const total_sum = _.reduce(
-        current_year_data,
+      const total_sum_tp = _.reduce(
+        current_year_data_tp,
         (sum, value) => sum += value,
         0
       );
-      const percent_of_total = current_year_data[largest_prov] / total_sum;
-      const text_args = {
-        largest_prov: provinces_with_article[largest_prov].text,
-        total_sum: formatter(total_sum),
-        percent_of_total: formats["percentage1_raw"](percent_of_total),
+      const percent_of_total_tp = current_year_data_tp[largest_prov_tp] / total_sum_tp;
+      const text_args_tp = {
+        largest_prov: provinces_with_article[largest_prov_tp].text,
+        total_sum: formatter(total_sum_tp),
+        percent_of_total: formats["percentage1_raw"](percent_of_total_tp),
         subject: calculations.subject,
+        show_per_capita: false,
+      };
+
+      const should_tab_be_disabled = _.every(data_tppc, _.isEmpty);
+      
+      //TP PER CAPITA VERSION
+      const current_year_data_tppc = _.last(data_tppc);
+      
+      const max_tppc = _.chain(data_tppc)
+        .last()
+        .values()
+        .max()
+        .value();
+      const color_scale_tppc = d3.scaleLinear()
+        .domain([0, max_tppc])
+        .range([0.2, 1]);
+
+      const largest_prov_tppc = _.chain(current_year_data_tppc)
+        .keys()
+        .maxBy( (prov) => current_year_data_tppc[prov] )
+        .value();
+      const total_sum_tppc = _.reduce(
+        current_year_data_tppc,
+        (sum, value) => sum += value,
+        0
+      );
+      const percent_of_total_tppc = current_year_data_tppc[largest_prov_tppc] / total_sum_tppc;
+      const text_args_tppc = {
+        largest_prov: (should_tab_be_disabled) ? null : provinces_with_article[largest_prov_tppc].text,
```

Comment:
don't actually need brackets around `should_tab_be_disabled`

Reply:
I think you could also write this as `should_tab_be_disabled && provinces_with_article[largest_prov_tppc].text`?

Reply:
!should_tab_be_disabled && provinces_with_article[largest_prov_tppc].text,
but yeah it would be more efficient
fixed 👍